encoder/vaapiencoder_base.cpp

```diff
@@ -498,7 +661,7 @@ bool VaapiEncoderBase::initVA()
 
     int32_t surfaceWidth = ALIGN16(m_videoParamCommon.resolution.width);
     int32_t surfaceHeight = ALIGN16(m_videoParamCommon.resolution.height);
-    m_pool = SurfacePool::create(m_alloc, YAMI_FOURCC_NV12, (uint32_t)surfaceWidth, (uint32_t)surfaceHeight, m_maxOutputBuffer);
+    m_pool = SurfacePool::create(m_alloc, ((m_videoParamCommon.bitDepth == 8)?YAMI_FOURCC_NV12:YAMI_FOURCC_P010), (uint32_t)surfaceWidth, (uint32_t)surfaceHeight, m_maxOutputBuffer);
```

Comment:
feed the VA_FOURCC_P016 when 10bit
use VA_FOURCC_NV12 as the default mode when bitDepth !=8

and if we add 12bit or more ,just add here in the later